objectherkenning_openbare_ruimte/data_sampling/decos_sampling.py

```diff
@@ -0,0 +1,306 @@
+
+            # TODO do we want to use gps_time instead?
+            meta_gdf = DecosSampling.load_metadata_gdf(meta_path).set_index(
+                "new_frame_id"
+            )[["timestamp", "geometry"]]
+            meta_gdf["timestamp"] = pd.to_datetime(meta_gdf["timestamp"], unit="s")
+            meta_data = meta_gdf.to_numpy()
+
+            # In case of a difference in number of rows, we clip or append None
+            diff = len(idx) - len(meta_data)
+            if diff > 0:
+                meta_data = np.vstack([meta_data, [[None, None]] * diff])
+            elif diff < 0:
+                meta_data = meta_data[0 : len(idx), :]
+
+            frame_gdf.loc[idx, ["timestamp", "geometry"]] = meta_data
+
+            # match frame locations with decos data
+            decos_bb = sg.box(*frame_gdf.loc[idx, :].total_bounds)
+            date = frame_gdf.loc[idx[0], "timestamp"]
+            decos_gdf_filtered = decos_helper.filter_decos_by_date(decos_gdf, date)
+            decos_clipped = decos_gdf_filtered.clip(decos_bb.buffer(decos_buffer))
+            if len(decos_clipped) >= 1:
+                decos_area = decos_clipped.unary_union.buffer(decos_buffer)
+                permits = frame_gdf.loc[idx, :].within(decos_area).to_numpy()
+                frame_gdf.loc[idx, "permit"] = permits
+
+            print(
+                f"{np.count_nonzero(frame_gdf.loc[idx, 'permit'])}/{len(frame_gdf.loc[idx, 'permit'])} frames in permit zone"
+            )
+
+        return frame_gdf
+
+    @staticmethod
+    def create_frame_gdf(
+        input_folder: str,
+    ) -> gpd.GeoDataFrame:
+        """
+        Create a GeoDataFrame to hold metadata information for all image frames
+        in a given folder.
+
+        Parameters
+        ----------
+        input_folder: str
+            Path to folder with image frames
+
+        Returns
+        -------
+        GeoDataFrame to hold the metadata, one row per image frame.
+        """
+        frames = sorted(list(pathlib.Path(input_folder).glob("*.jpg")))
+
+        frame_data = {
+            "path": frames,
+            "frame_src": [
+                frame.name.replace(".jpg", "").rsplit(sep="_", maxsplit=1)[0]
+                for frame in frames
+            ],
+            "frame_id": [
+                int(frame.name.replace(".jpg", "").rsplit(sep="_", maxsplit=1)[1])
+                for frame in frames
+            ],
+            "geometry": np.empty_like(frames, dtype=object),
+            "timestamp": np.empty_like(frames, dtype=object),
+            "permit": np.zeros_like(frames, dtype=bool),
+            "sample": np.zeros_like(frames, dtype=bool),
+        }
+        frame_gdf = gpd.GeoDataFrame(data=frame_data, crs=RD_CRS)
+
+        print(f"{len(frame_gdf)} frames found")
+
+        return frame_gdf
+
+    def _sample_frames(self):
+        """
+        Sample frames based on sampling weight and permit data.
+        """
```

Comment:
The docstring could be more informative, for example it could explain what happens during the weights selection based on the value of permit, and what happens in the two cases of the if-else statement